tests/s3/test_s3_artifacts.py

```diff
@@ -0,0 +1,225 @@
+import pytest
+from unittest.mock import MagicMock, patch
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from hashlib import md5, sha256
+from gardenlinux.s3.s3_artifacts import S3Artifacts
+
+
+# Dummy CName replacement
+class DummyCName:
+    def __init__(self, cname):
+        self.platform = "aws"
+        self.arch = "amd64"
+        self.version = "1234.1"
+        self.commit_id = "abc123"
+
+
+# Helpers to compute digests for fake files
+def dummy_digest(data: bytes, algo: str) -> str:
+    """
+    Dummy for file_digest() to compute hashes for in-memory byte streams
+    """
+    content = data.read()
+    data.seek(0)  # Reset byte cursor to start for multiple uses
+
+    if algo == "md5":
+        return md5(content)  # nosec B324
+    elif algo == "sha256":
+        return sha256(content)
+    else:
+        raise ValueError(f"Unsupported algo: {algo}")
+
+
+@patch("gardenlinux.s3.s3_artifacts.Bucket")
```

Comment:
I do like the simplicity of the `mock_aws` decorator used in `tests/s3/test_bucket.py`. Do you think we could introduce an `setup` and `teardown` (if needed) function to fill the mocked bucket with well-known data instead of hacking together a mocked `gardenlinux.s3.bucket.Bucket` variant here?

As far as I see the way `mock_aws` works the `S3Artifacts` underlying `Bucket` instance should still be easily able to handle the mocked AWS resources and the `mock_aws` decorator has been successfully verified with `tests/s3/test_bucket.py`.

Reply:
I have found an even cleaner solution for the `S3Artifacts` tests: Using `mock_aws` as context handler in a fixture. This will avoid the repeated manual setup of a mock bucket that currently exists in `test_bucket.py`.

I will see if I can apply this change to `test_bucket.py` as well.